### ISSUE TITLE
GNS 430 Hardware Plate Dimensions

### ISSUE THREAD

**Issue by acirinelli** [allanglen/c172-flight-sim-panel#12]
The [uxcell mini push buttons](https://www.amazon.com/gp/product/B00OK476IG/ref=as_li_ss_tl?creativeASIN=B00OK476IG&imprToken=gMRGvPVcPTGMYbWErl9k0g&slotNum=1&ie=UTF8&linkCode=w61&tag=c172-20&linkId=5fd3ae40e8fae5b1117cc293f6f73be1) have a dimension of 7mm (~0.273 in) and I cannot get them to fit in a 3D printed hardware plate which has dimensions closer to 0.25 in when finished. I tried modifying the cad file to 0.27 in and it still prints at 0.265 in. I may up the size again another 100th and see how it prints, however...

The SVG file has dimensions of 0.2618 in. When this part was laser cut, did the 7mm push buttons fit snug in that hole and was that the actual cut dimension? Is it possible that the dimensions changed slightly since you purchased the push buttons? I can't get the buttons to fit in a printed 0.265 in hole, so it doesn't make sense to me.

I don't want to pay for laser cut parts that may not work (which is why I am trying to print them). Any feedback would be helpful.

**Comment by acirinelli** [comment]
For anyone interested, I increased the button dimensions to 0.28 which printed at about 0.269-0.272. This may vary printer to printer. The buttons fit snug and I had to sand a few holes a little bit for nice smooth fit and button press. Then I used dabs of hot glue on the back side to ensure they would stay in place.

**EDIT: I recently discovered my 3D printer was out of tolerance and spent many hours trying to dial it in. I'm sure I could print more accurately now.** However, the buttons dimensions were still different than the model dimensions and you may have to adjust.

![IMG_3350](https://user-images.githubusercontent.com/3673934/78463982-a98e4000-7698-11ea-80a0-fb25a33ca4f4.JPG)
![IMG_3349](https://user-images.githubusercontent.com/3673934/78463985-b1e67b00-7698-11ea-937c-94dc660f68d2.JPG)

**Comment by acirinelli** [comment]
I also added a cross beam piece to support 2 multiplexers for all the switches in the unit.
![89A9DCD1-0372-474A-ABDF-A2774D231988](https://user-images.githubusercontent.com/3673934/78622120-26f0b680-7839-11ea-968d-b8bb8ec408b1.jpeg)

**Comment by acirinelli** [comment]
I used the [Propwash dual encoder](https://www.propwashsim.com/store/dual-encoder-kit) and the outer/inner buttons did not fit, very loose and just fell off. The dimensions of the encoder shafts were fairly different than the model dimensions. I revised the parts and stl files, but I am unable to share here. Just an FYI for anyone trying to print with this encoder, it may not work for you.

Note: You don't necessarily need the printed knobs, it just adds to the realism and makes it a little easier to control because they are larger diameters. The dual encoder comes with knobs (although depending on faceplate thickness, like mine, you may not be able to use the button press functionality).

**Comment by JTRACKER** [comment]
Does anyone have the part number for the encoder being used?  I don't want to pay >$20 for a PCB/encoder/knobs - if I don't have to.

**Comment by acirinelli** [comment]
> Does anyone have the part number for the encoder being used? I don't want to pay >$20 for a PCB/encoder/knobs - if I don't have to.

No, but here is an [updated link](https://www.propwashsim.com/store/dual-encoder-kit) to a cheaper version direct from Propwash. Best bet is to reach out to them.

**Comment by Austinthemighty** [comment]
I got those ones from prop wash but struggled to get them to sit right, patently they have a square hole, which made it quite difficult.

**Comment by acirinelli** [comment]
> I got those ones from prop wash but struggled to get them to sit right, patently they have a square hole, which made it quite difficult.

I believe you are referring to the notch on them. I took a metal file to that and filed it off. Sits flush on the back side into a round hole.

**Comment by Austinthemighty** [comment]
So I did remove the notch on mine but the problem was that the threads were not long enough on the encoder to fit through the acrylic

**Comment by rubicon0007** [comment]
> For anyone interested, I increased the button dimensions to 0.28 which printed at about 0.269-0.272. This may vary printer to printer. The buttons fit snug and I had to sand a few holes a little bit for nice smooth fit and button press. Then I used dabs of hot glue on the back side to ensure they would stay in place.
> 
> **EDIT: I recently discovered my 3D printer was out of tolerance and spent many hours trying to dial it in. I'm sure I could print more accurately now.** However, the buttons dimensions were still different than the model dimensions and you may have to adjust.
> 
> ![IMG_3350](https://user-images.githubusercontent.com/3673934/78463982-a98e4000-7698-11ea-80a0-fb25a33ca4f4.JPG) ![IMG_3349](https://user-images.githubusercontent.com/3673934/78463985-b1e67b00-7698-11ea-937c-94dc660f68d2.JPG)

I had the same issue with the 530. I've currently got 3 full prints and two half prints before finding the proper button size and spacing. My printer does a great job printing smooth lines without excessive squish or any other problems so i know it had to be an issue between printing and laser cutting. My buttons are 7mm as well and increased the button holes to 7.2mm using tinkercad. Finally, I've made enough progress to actually get good results. I'll provide a thingiverse link later with the edited STL so more people dont run into the same issue. I could not find a proper file that would allow for the buttons to fit the right way. NO WAY was i going to file out all of these holes!
